### ISSUE TITLE
Subscription field got Hash object

### ISSUE THREAD

**Issue by cpunion** [rmosolgo/graphql-ruby#969]
I just followed the guide about subscriptions, at the step "define subscription type" (http://graphql-ruby.org/subscriptions/subscription_type.html), there has an exception raised, because the `postWasPublished` field got Hash object, but `Types::PostType` accepts `Post` model object. I must modify to:

```ruby
# app/graphql/types/subscription_type.rb
Types::SubscriptionType = GraphQL::ObjectType.define do
  name "Subscription"
  field :postWasPublished, !Types::PostType, "A post was published to the blog" do
    resolve ->(obj, args, ctx) {
      Post.new(obj)
    }
  end
  # ...
end
```

It's not clear.

**Comment by cpunion** [comment]
Except for this, it works find!

Thank you for this awesome feature.

**Comment by cpunion** [comment]
I just read the code of ActionCable and ActiveJob, they are both serialize/deserialize `GlobalID::Identification` types to keep model object's type. When I pass a model object to `Schema.subscriptions.trigger`, I think we can do the same thing.

**Comment by rmosolgo** [comment]
fixed by #1004
